kratos/utilities/parallel_utilities.cpp

```diff
@@ -106,22 +106,29 @@ int ParallelUtilities::InitializeNumberOfThreads()
 #endif
 }
 
-int& ParallelUtilities::GetNumberOfThreads()
+int ParallelUtilities::GetNumberOfThreads()
 {
-    if (!mspNumThreads) {
-        LockObject lock;
-        lock.SetLock();
-        if (!mspNumThreads) {
-            static int num_threads;
-            num_threads = InitializeNumberOfThreads();
-            mspNumThreads = &num_threads;
-        }
-        lock.UnSetLock();
-    }
-
-    return *mspNumThreads;
+    // if (!mspNumThreads) {
+    //     LockObject lock;
+    //     lock.SetLock();
+    //     if (!mspNumThreads) {
+    //         static int num_threads;
+    //         num_threads = InitializeNumberOfThreads();
+    //         mspNumThreads = &num_threads;
+    //     }
+    //     lock.UnSetLock();
+    // }
+
+    // return *mspNumThreads;
+#ifdef KRATOS_SMP_OPENMP
+    return omp_get_num_threads();
```

Comment:
I am ok with using this, but why change the behavior completly even if we don't use OpenMP?

Reply:
To be honest I don't know how to keep the new behaviour only if we don't use OpenMP.

This is why only I marked this as a draft, the code as it is now solves the bug but I don't know if its the better approach.